### ISSUE TITLE
RPG fires when trying to point to something

### ISSUE THREAD

**Issue by dasWurmtMich** [tgstation/TerraGov-Marine-Corps#337]
## Reproduction:

try to point to someone or something and watch it shoot the RPG (shift middlemousebutton)

**Comment by Ghommie** [comment]
Couldn't reproduce. There is an off chance you misclicked the left button and stood still for a moment, I saw no immediate feed back upon the click, and the busy circling icon isn't very noticeable when trying it.

**Comment by Monikawales** [comment]
Also tank fires NOTHING with flamer when trying to point-fire alien body

**Comment by Monikawales** [comment]
Sometimes tank fires cannon when click interface buttons

**Comment by dasWurmtMich** [comment]
> 
> 
> Couldn't reproduce. There is an off chance you misclicked the left button and stood still for a moment, as there is immediate feed back upon the click, and the busy wheel icon isn't very noticeable.

I  literally reproduced it myself after misfiring the first time via shift+MMB, I did have a miniscope so it might be because of that

**Comment by DominikPanic** [comment]
@dasWurmtMich I was able to replicate - if you try to point WHILE scope it, you shoot.
@Ghommie

**Comment by DominikPanic** [comment]
It's because you are literally unable to point since how zooming works is that it changes your viewpoint.

**Comment by DominikPanic** [comment]
For anyone that wants to pursue this further, these are my notes, seems like whenever your viewpoint changes you are not longer a carbon mob.

```
What happens when you middle or shift+middle click while zoomed in:
Client click called
Control, passing to do_click
Mob Do_click called
Atom clicked called


What should happen instead (shift+middle click while not zoomed in)
Client click called
Control, passing to do_click
Mob Do_click called
carbon click called
Pointing detected, pointing
Jane Doe points to the floor
Atom clicked called
Click handled elsewhere, returning
```

**Comment by DominikPanic** [comment]
Figured it out, blame set_interaction.
Fixed in #459
